pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -116,7 +117,7 @@ func (r *Reconciler) getOrCreateClusterPolicySnapshot(ctx context.Context, crp *
 	}
 
 	// latestPolicySnapshotIndex should be -1 when there is no snapshot.
-	latestPolicySnapshot, latestPolicySnapshotIndex, err := r.lookupLatestClusterPolicySnapshot(ctx, crp)
+	latestPolicySnapshot, latestPolicySnapshotIndex, sortedList, err := r.lookupLatestClusterPolicySnapshot(ctx, crp)
```

Comment:
Considering that the cleanup is not a very frequent op, would it be better to do it when the size limit is actually over rather than doing it for each list?

Reply:
The list will only happen when it needs to create an new snapshot. Removed this return to clear the confusion.